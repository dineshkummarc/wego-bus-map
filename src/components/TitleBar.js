import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom';
import { faRoute, faInfoCircle } from '@fortawesome/free-solid-svg-icons';
import { faGithub } from '@fortawesome/free-brands-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import logo from '../resources/logo.svg';

function TitleBar({hide}) {
  if (hide) {
    return
  }
  return(
    <div>
      <div className="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 navbar-dark bg-dark border-bottom shadow-sm">
          <Link to="/" className="text-light h3"><img src={logo} width="30" height="30" className="d-inline-block align-top me-2" alt="Bus" /> WeGo Bus Map</Link>
        <div className="h5 my-0 me-md-auto">
        </div>
        <nav className="my-2 my-md-0 mr-md-3">
          <Link className="p-2 text-light" to="/about"><FontAwesomeIcon icon={faInfoCircle} fixedWidth={true} /> About</Link>
          <Link className="p-2 text-light" to="/routes"><FontAwesomeIcon icon={faRoute} fixedWidth={true} /> Routes</Link>
          <Link className="p-2 text-light" to="https://github.com/transitnownash/wego-bus-map" target="_blank"><FontAwesomeIcon icon={faGithub} fixedWidth={true} /> View on GitHub</Link>
        </nav>
        <Link className="btn btn-outline-light" to="/">Return to App</Link>
      </div>
    </div>
  )
}

TitleBar.propTypes = {
  hide: PropTypes.bool
}

export default TitleBar
