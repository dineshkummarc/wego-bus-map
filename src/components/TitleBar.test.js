/* globals test, expect */

import React from 'react';
import TitleBar from './TitleBar';
import { render } from '@testing-library/react';
import { BrowserRouter as Router} from 'react-router-dom';

test('renders TitleBar', () => {
  const {container} = render(
    <Router>
      <TitleBar />
    </Router>
  );
  expect(container).toMatchSnapshot();
});
