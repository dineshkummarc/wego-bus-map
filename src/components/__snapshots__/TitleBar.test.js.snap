// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TitleBar 1`] = `
<div>
  <div>
    <div
      class="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 navbar-dark bg-dark border-bottom shadow-sm"
    >
      <a
        class="text-light h3"
        href="/"
      >
        <img
          alt="Bus"
          class="d-inline-block align-top me-2"
          height="30"
          src="logo.svg"
          width="30"
        />
         WeGo Transit Map
      </a>
      <div
        class="h5 my-0 me-md-auto"
      />
      <nav
        class="my-2 my-md-0 mr-md-3"
      >
        <a
          class="p-2 text-light"
          href="/about"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-info fa-fw "
            data-icon="circle-info"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 0C114.6 0 0 114.6 0 256s114.6 256 256 256s256-114.6 256-256S397.4 0 256 0zM256 128c17.67 0 32 14.33 32 32c0 17.67-14.33 32-32 32S224 177.7 224 160C224 142.3 238.3 128 256 128zM296 384h-80C202.8 384 192 373.3 192 360s10.75-24 24-24h16v-64H224c-13.25 0-24-10.75-24-24S210.8 224 224 224h32c13.25 0 24 10.75 24 24v88h16c13.25 0 24 10.75 24 24S309.3 384 296 384z"
              fill="currentColor"
            />
          </svg>
           About
        </a>
        <a
          class="p-2 text-light"
          href="/routes"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-route fa-fw "
            data-icon="route"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320 256C302.3 256 288 270.3 288 288C288 305.7 302.3 320 320 320H416C469 320 512 362.1 512 416C512 469 469 512 416 512H139.6C148.3 502.1 158.9 489.4 169.6 475.2C175.9 466.8 182.4 457.6 188.6 448H416C433.7 448 448 433.7 448 416C448 398.3 433.7 384 416 384H320C266.1 384 223.1 341 223.1 288C223.1 234.1 266.1 192 320 192H362.1C340.2 161.5 320 125.4 320 96C320 42.98 362.1 0 416 0C469 0 512 42.98 512 96C512 160 416 256 416 256H320zM416 128C433.7 128 448 113.7 448 96C448 78.33 433.7 64 416 64C398.3 64 384 78.33 384 96C384 113.7 398.3 128 416 128zM118.3 487.8C118.1 488 117.9 488.2 117.7 488.4C113.4 493.4 109.5 497.7 106.3 501.2C105.9 501.6 105.5 502 105.2 502.4C99.5 508.5 96 512 96 512C96 512 0 416 0 352C0 298.1 42.98 255.1 96 255.1C149 255.1 192 298.1 192 352C192 381.4 171.8 417.5 149.9 448C138.1 463.2 127.7 476.9 118.3 487.8L118.3 487.8zM95.1 384C113.7 384 127.1 369.7 127.1 352C127.1 334.3 113.7 320 95.1 320C78.33 320 63.1 334.3 63.1 352C63.1 369.7 78.33 384 95.1 384z"
              fill="currentColor"
            />
          </svg>
           Routes
        </a>
        <a
          class="p-2 text-light"
          href="/bcycle"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bicycle fa-fw "
            data-icon="bicycle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M347.2 32C358.1 32 369.8 38.44 375.4 48.78L473.3 229.1C485.5 226.1 498.5 224 512 224C582.7 224 640 281.3 640 352C640 422.7 582.7 480 512 480C441.3 480 384 422.7 384 352C384 311.1 402.4 276.3 431.1 252.8L409.4 212.7L324.7 356.2C320.3 363.5 312.5 368 304 368H255C247.1 431.1 193.3 480 128 480C57.31 480 0 422.7 0 352C0 281.3 57.31 224 128 224C138.7 224 149.2 225.3 159.2 227.8L185.8 174.7L163.7 144H120C106.7 144 96 133.3 96 120C96 106.7 106.7 96 120 96H176C183.7 96 190.1 99.71 195.5 105.1L222.9 144H372.3L337.7 80H311.1C298.7 80 287.1 69.25 287.1 56C287.1 42.75 298.7 32 311.1 32H347.2zM440 352C440 391.8 472.2 424 512 424C551.8 424 584 391.8 584 352C584 312.2 551.8 280 512 280C508.2 280 504.5 280.3 500.8 280.9L533.1 340.6C539.4 352.2 535.1 366.8 523.4 373.1C511.8 379.4 497.2 375.1 490.9 363.4L458.6 303.7C447 316.5 440 333.4 440 352V352zM108.8 328.6L133.1 280.2C131.4 280.1 129.7 280 127.1 280C88.24 280 55.1 312.2 55.1 352C55.1 391.8 88.24 424 127.1 424C162.3 424 190.9 400.1 198.2 368H133.2C112.1 368 99.81 346.7 108.8 328.6H108.8zM290.3 320L290.4 319.9L217.5 218.7L166.8 320H290.3zM257.4 192L317 274.8L365.9 192H257.4z"
              fill="currentColor"
            />
          </svg>
           BCycle
        </a>
      </nav>
    </div>
  </div>
</div>
`;
